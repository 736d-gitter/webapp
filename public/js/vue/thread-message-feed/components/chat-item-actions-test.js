const mount = require('../../__test__/vuex-mount');
const { default: ChatItemActions } = require('./chat-item-actions.vue');
const { createSerializedMessageFixture } = require('../../__test__/fixture-helpers');

jest.mock('gitter-web-client-context');
const context = require('gitter-web-client-context');

describe('thread-message-feed chat-item-actions', () => {
  const message = createSerializedMessageFixture();

  beforeEach(() => {
    context.mockReset();
    context.mockImplementation(() => ({}));
  });

  it('matches snapshot', () => {
    const { wrapper } = mount(ChatItemActions, { message });
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('user can delete the message', () => {
    beforeEach(() => {
      context.getUserId.mockImplementation(function() {
        return message.fromUser.id;
      });
    });

    it('matches snapshot', () => {
      const { wrapper } = mount(ChatItemActions, { message });
      expect(wrapper.element).toMatchSnapshot();
    });

    it('triggers delete action when delete option is clicked', () => {
      const { wrapper, stubbedActions } = mount(ChatItemActions, { message });
      // removing the original implementation to prevent an API call
      stubbedActions.threadMessageFeed.deleteMessage.mockImplementation(() => {});
      wrapper.find('.js-chat-item-delete-action').trigger('click');
      expect(stubbedActions.threadMessageFeed.deleteMessage).toHaveBeenCalled();
    });
  });

  describe('user can edit the message', () => {
    const testMessage = createSerializedMessageFixture({ sent: new Date().toISOString() });
    beforeEach(() => {
      context.getUserId.mockImplementation(function() {
        return testMessage.fromUser.id;
      });
    });

    it('matches snapshot', () => {
      const { wrapper } = mount(ChatItemActions, { message: testMessage });
      expect(wrapper.element).toMatchSnapshot();
    });

    it('triggers edit action when edit option is clicked', () => {
      const { wrapper, stubbedActions } = mount(ChatItemActions, { message: testMessage });
      // removing the original implementation to prevent an API call
      stubbedActions.threadMessageFeed.editMessage.mockImplementation(() => {});
      wrapper.find('.js-chat-item-edit-action').trigger('click');
      expect(stubbedActions.threadMessageFeed.editMessage).toHaveBeenCalled();
    });
  });
});
