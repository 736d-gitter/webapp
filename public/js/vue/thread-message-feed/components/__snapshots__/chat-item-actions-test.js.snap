// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`thread-message-feed chat-item-actions matches snapshot 1`] = `
<div
  class="chat-item__actions"
>
  <b-popover-stub
    boundary="scrollParent"
    boundarypadding="5"
    customclass="chat-item-actions-popover"
    delay="0"
    fallbackplacement="flip"
    offset="0"
    placement="left"
    target="chat-actions-5d147ea84dad9dfbc522317a"
    title=""
    triggers="click blur"
  >
    <div
      class="popover-item__action-disabled"
    >
      
      Edit
    
    </div>
     
    <div
      class="popover-item__action-disabled"
    >
      Delete
    </div>
  </b-popover-stub>
   
  <button
    class="chat-item-actions-button"
    id="chat-actions-5d147ea84dad9dfbc522317a"
  >
    <i
      class="chat-item__icon icon-ellipsis"
    />
  </button>
</div>
`;

exports[`thread-message-feed chat-item-actions user can delete the message matches snapshot 1`] = `
<div
  class="chat-item__actions"
>
  <b-popover-stub
    boundary="scrollParent"
    boundarypadding="5"
    customclass="chat-item-actions-popover"
    delay="0"
    fallbackplacement="flip"
    offset="0"
    placement="left"
    target="chat-actions-5d147ea84dad9dfbc522317a"
    title=""
    triggers="click blur"
  >
    <div
      class="popover-item__action-disabled"
    >
      
      Edit
    
    </div>
     
    <button
      class="popover-item__action js-chat-item-delete-action"
      title="Delete this message"
    >
      
      Delete
    
    </button>
  </b-popover-stub>
   
  <button
    class="chat-item-actions-button"
    id="chat-actions-5d147ea84dad9dfbc522317a"
  >
    <i
      class="chat-item__icon icon-ellipsis"
    />
  </button>
</div>
`;

exports[`thread-message-feed chat-item-actions user can edit the message matches snapshot 1`] = `
<div
  class="chat-item__actions"
>
  <b-popover-stub
    boundary="scrollParent"
    boundarypadding="5"
    customclass="chat-item-actions-popover"
    delay="0"
    fallbackplacement="flip"
    offset="0"
    placement="left"
    target="chat-actions-5d147ea84dad9dfbc522317a"
    title=""
    triggers="click blur"
  >
    <button
      class="popover-item__action js-chat-item-edit-action"
      title="Edit this message"
    >
      
      Edit
    
    </button>
     
    <button
      class="popover-item__action js-chat-item-delete-action"
      title="Delete this message"
    >
      
      Delete
    
    </button>
  </b-popover-stub>
   
  <button
    class="chat-item-actions-button"
    id="chat-actions-5d147ea84dad9dfbc522317a"
  >
    <i
      class="chat-item__icon icon-ellipsis"
    />
  </button>
</div>
`;
